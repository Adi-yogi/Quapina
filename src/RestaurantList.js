import React from 'react';
import { connect } from 'react-redux';
import { LIST_STATUS } from 'store/reducer.js';

export const RestaurantList = ({ items = [], filterBy = undefined, searchInCity = '', listStatus }) => {

    const filterItems = () => items.filter(item => {
        const { name, address, area } = item;
        const searchIn = `${name}${address}${area}`.toLowerCase();
        return searchIn.toLowerCase().includes(filterBy.toLowerCase());
    });
    const displayItems = !!filterBy ? filterItems() : items;

    if (listStatus === LIST_STATUS.INITIAL) {
        return <div className="incenter">
            Please enter a city to search. e.g. Orleans,Osseo,Norman etc.
        </div>
    }
    else if (listStatus === LIST_STATUS.FETCHING) {
        return <div className="incenter">
            Please wait
        </div>
    } else if (displayItems.length === 0) {
        return <div className="incenter">
            No Restaurant found for {searchInCity} {!!filterBy && `and ${filterBy}`}
        </div >
    }
    return <React.Fragment>
        <div className="resultHeader">
            <h2>Name</h2>
            <h2>Address</h2>
            <h2>Price</h2>
        </div>
        <div className="results">
            {displayItems.map(item => <div className="resultElement" key={item.id}>
                <h3>{item.name}</h3>
                <h3>{item.address}</h3>
                <h3>{item.price}</h3>
            </div>)}
        </div>
    </React.Fragment>
}

const mapStateToProps = ({ restaurant }) => ({
    items: restaurant.items,
    searchInCity: restaurant.searchInCity,
    filterBy: restaurant.filterBy,
    listStatus: restaurant.listStatus
});

export default connect(mapStateToProps)(RestaurantList)