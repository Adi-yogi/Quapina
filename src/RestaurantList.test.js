import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantList } from './RestaurantList.js';
import { LIST_STATUS } from 'store/reducer.js';


it('renders without crashing', () => {
    const props = {
        listStatus: LIST_STATUS.INITIAL
    }
    const restaurantList = shallow(<RestaurantList {...props} />);
    expect(restaurantList).toMatchSnapshot();
});

it('renders properly when no items', () => {
    const props = {
        listStatus: LIST_STATUS.FETCH_DONE,
        items: [],
        searchInCity: 'Boston'
    }
    const restaurantList = shallow(<RestaurantList {...props} />);
    expect(restaurantList).toMatchSnapshot();
});

it('renders properly when filter results no items', () => {
    const props = {
        listStatus: LIST_STATUS.FETCH_DONE,
        items: [],
        searchInCity: 'Boston',
        filterBy: 'kitchen'
    }
    const restaurantList = shallow(<RestaurantList {...props} />);
    expect(restaurantList).toMatchSnapshot();
});
it('renders properly when loading', () => {
    const props = {
        listStatus: LIST_STATUS.FETCHING
    }
    const restaurantList = shallow(<RestaurantList {...props} />);
    expect(restaurantList).toMatchSnapshot();
});

it('renders properly with data', () => {
    const props = {
        listStatus: LIST_STATUS.FETCH_DONE,
        items: [
            {
                id: 1635,
                name: 'Elephant Walk, The - Boston',
                address: '900 Beacon Street',
                area: 'Boston / New England',
                price: 2
            },
            {
                id: 2829,
                name: 'Metropolis Cafe',
                address: '584 Tremont Street',
                area: 'Boston / New England',
                price: 2
            }
        ]
    }
    const restaurantList = shallow(<RestaurantList {...props} />);
    expect(restaurantList).toMatchSnapshot();
});

it('renders properly when filtering', () => {
    const props = {
        listStatus: LIST_STATUS.FETCH_DONE,
        items: [
            {
                id: 1635,
                name: 'Elephant Walk, The - Boston',
                address: '900 Beacon Street',
                area: 'Boston / New England',
                price: 2
            },
            {
                id: 2829,
                name: 'Metropolis Cafe',
                address: '584 Tremont Street',
                area: 'Boston / New England',
                price: 2
            },
            {
                id: 2642,
                name: 'Troquet',
                address: '140 Boylston Street',
                area: 'Boston / New England',
                price: 3
            },
            {
                id: 5907,
                name: 'Terramia',
                address: '98 Salem St., North End',
                area: 'Boston / New England',
                price: 3
            },
            {
                id: 2827,
                name: 'Rowes Wharf Sea Grille',
                address: '70 Rowes Wharf',
                area: 'Boston / New England',
                price: 2
            }
        ],
        filterBy: 'Rowes'
    }
    const restaurantList = shallow(<RestaurantList {...props} />);
    expect(restaurantList).toMatchSnapshot();
});
