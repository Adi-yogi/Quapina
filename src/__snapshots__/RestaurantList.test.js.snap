// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly when filter results no items 1`] = `
<div
  className="incenter"
>
  No Restaurant found for 
  Boston
   
  and kitchen
</div>
`;

exports[`renders properly when filtering 1`] = `
<Fragment>
  <div
    className="resultHeader"
  >
    <h2>
      Name
    </h2>
    <h2>
      Address
    </h2>
    <h2>
      Price
    </h2>
  </div>
  <div
    className="results"
  >
    <div
      className="resultElement"
      key="2827"
    >
      <h3>
        Rowes Wharf Sea Grille
      </h3>
      <h3>
        70 Rowes Wharf
      </h3>
      <h3>
        2
      </h3>
    </div>
  </div>
</Fragment>
`;

exports[`renders properly when loading 1`] = `
<div
  className="incenter"
>
  Please wait
</div>
`;

exports[`renders properly when no items 1`] = `
<div
  className="incenter"
>
  No Restaurant found for 
  Boston
   
</div>
`;

exports[`renders properly with data 1`] = `
<Fragment>
  <div
    className="resultHeader"
  >
    <h2>
      Name
    </h2>
    <h2>
      Address
    </h2>
    <h2>
      Price
    </h2>
  </div>
  <div
    className="results"
  >
    <div
      className="resultElement"
      key="1635"
    >
      <h3>
        Elephant Walk, The - Boston
      </h3>
      <h3>
        900 Beacon Street
      </h3>
      <h3>
        2
      </h3>
    </div>
    <div
      className="resultElement"
      key="2829"
    >
      <h3>
        Metropolis Cafe
      </h3>
      <h3>
        584 Tremont Street
      </h3>
      <h3>
        2
      </h3>
    </div>
  </div>
</Fragment>
`;

exports[`renders without crashing 1`] = `
<div
  className="incenter"
>
  Please enter a city to search. e.g. Orleans,Osseo,Norman etc.
</div>
`;
